lib/auth/auth.ts

```diff
@@ -181,7 +181,8 @@ function generateV4Headers(
     secretKeyValue: string,
     awsService: string,
     proxyPath: string,
-    sessionToken: string
+    sessionToken: string,
+    payload: string,
```

Comment:
```suggestion
    payload: string?,
```

Also can you add the JSDoc entry for this param